doc/manual/rl-next/flake-add-git-ignored-files.md

```diff
@@ -0,0 +1,28 @@
+---
+synopsis: Nix no longer attempts to `git add` files that are `.gitignore`d
+prs: 9324
+issues: 8854
+---
+
+Previously, most flake operations (like `nix build`, `nix develop`, or `nix
+flake metadata`) in a Git repository where `flake.lock` was `.gitignore`d would
+fail:
+
+```
+$ nix develop
+'The following paths are ignored by one of your .gitignore files:
+config/home-mangler/flake.lock
+hint: Use -f if you really want to add them.
+hint: Turn this message off by running
+hint: "git config advice.addIgnoredFile false"
+error:
+       … while updating the lock file of flake 'git+file:///Users/wiggles/.dotfiles?dir=config/home-mangler&ref=refs/heads/main&rev=f4d806103ef76becbc220c8e3d09bf6ac1e78bb8'
+
+       error: program 'git' failed with exit code 1
+```
+
+Now, Nix will detect that the file is `.gitignore`d and skip adding it, making
+commands like `nix develop` succeed.
+
+This was actually released in Nix 2.20, but wasn't added to the release notes
+so we're announcing it here. The historical release notes have been updated as well.
```

Comment:
This will cause this release notes entry to be duplicated for 2.20 and 2.21. I don't think we want that. So let's only include this in `rl-2.20.md`.

Reply:
Yep, I think it's fine to backport release notes manually for completeness though. We don't have to make a point release just for those.